torch/testing/asserts.py

```diff
@@ -0,0 +1,186 @@
+import functools
+import inspect
+import warnings
+from typing import Callable, Optional, Tuple
+
+import torch
+from torch.testing._internal.common_utils import get_comparison_dtype as _get_comparison_dtype, TestCase as _TestCase
+
+__all__ = ["assert_tensors_equal", "assert_tensors_allclose"]
+
+try:
+    import pytest
+
+    UsageError = pytest.UsageError
+except ImportError:
+
+    class UsageError(Exception):
+        pass
+
+
+class UsageWarning(Warning):
+    pass
+
+
+def _get_default_rtol_and_atol(tensor1: torch.Tensor, tensor2: torch.Tensor) -> Tuple[float, float]:
+    return _TestCase()._getDefaultRtolAndAtol(tensor1.dtype, tensor2.dtype)
+
+
+def _hide_internal_traceback(fn):
+    @functools.wraps(fn)
+    def wrapper(*args, **kwargs):
+        try:
+            return fn(*args, **kwargs)
+        except AssertionError:
+            for frame_info in inspect.trace():
+                frame_info.frame.f_locals.setdefault("__tracebackhide__", True)
+            raise
+
+    return wrapper
+
+
+# These attributes will be checked by default in every call to assert_tensors_(equal|allclose) and
+# assert_(equal|allclose) if the inputs are tensors.
+DEFAULT_CHECKED_ATTRIBUTES = (
+    "shape",
+    "dtype",
+)
+
+
+def _compare_tensors_meta(tensor1: torch.Tensor, tensor2: torch.Tensor, **attributes: bool) -> None:
+    if not (isinstance(tensor1, torch.Tensor) and isinstance(tensor2, torch.Tensor)):
+        raise AssertionError(f"Both inputs have to be tensors, but got {type(tensor1)} and {type(tensor2)} instead.")
+
+    for attr in DEFAULT_CHECKED_ATTRIBUTES:
```

Comment:
This approach to identifying which attributes to test seems fine for now. In the future it may be simpler to pass functions that operate on a single tensor instead to allow the computation of arbitrary properties.

Reply:
I don't understand what you mean here. Could you give me an example?

Reply:
```
(..., lambda t: t.device(), ...)
```